File: DataAccess/Models/Author.cs
using System.Text.Json.Serialization;

namespace DataAccess.Models;

public partial class Author
{
    public int Id { get; set; }

    public DateTime? AuthorBirthday { get; set; }

    public string? AuthorEmail { get; set; }

    public string? AuthorPhoneNumber { get; set; }

    public string? AuthorName { get; set; }

    [JsonIgnore]
    public virtual ICollection<Book> Books { get; } = new List<Book>();
}

File: WestCoastEducation/ClientApp/src/Models/author.ts
import { BookBriefModel } from "./book";

export interface IAuthorModel {
    id: number;
    authorBirthday?: Date;
    authorEmail?: string;
    authorPhoneNumber?: string;
    authorName?: string;
    books: BookBriefModel[];
}

export default class AuthorModel implements IAuthorModel {
    id: number;
    authorBirthday?: Date;
    authorEmail?: string;
    authorPhoneNumber?: string;
    authorName?: string;
    books: BookBriefModel[];

    constructor(props: IAuthorModel) {
        this.id = props.id;
        this.authorBirthday = props.authorBirthday;
        this.authorEmail = props.authorEmail;
        this.authorPhoneNumber = props.authorPhoneNumber;
        this.authorName = props.authorName;
        this.books = props.books;
    }
}